sdk/textanalytics/Azure.AI.TextAnalytics/src/OperationContinuationToken.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using Azure.Core;
+
+using ServiceVersion = Azure.AI.TextAnalytics.TextAnalyticsClientOptions.ServiceVersion;
+
+namespace Azure.AI.TextAnalytics
+{
+    internal class OperationContinuationToken
+    {
+        private const ServiceVersion ContinuationTokenVersion = ServiceVersion.V3_1;
+
+        // Parameterless constructor required for JSON deserialization.
+        public OperationContinuationToken()
+        {
+            Version = TextAnalyticsClientOptions.GetVersionString(ContinuationTokenVersion);
+        }
+
+        public OperationContinuationToken(string jobId, IDictionary<string, int> inputDocumentOrder, bool? showStats)
+            : this()
+        {
+            JobId = jobId;
+            InputDocumentOrder = new Dictionary<string, int>(inputDocumentOrder);
+            ShowStats = showStats;
+        }
+
+        public string Version { get; set; }
+
+        public string JobId { get; set; }
+
+        public Dictionary<string, int> InputDocumentOrder { get; set; }
+
+        public bool? ShowStats { get; set; }
+
+        /// <exception cref="ArgumentException">Thrown when the <see cref="Version"/> of the deserialized token is not the expected <see cref="ContinuationTokenVersion"/>.</exception>
+        /// <exception cref="FormatException">Thrown when <paramref name="base64OperationId"/> is not a valid base-64 string.</exception>
+        /// <exception cref="JsonException">Thrown when <paramref name="base64OperationId"/> cannot be deserialized from JSON.</exception>
+        public static OperationContinuationToken Deserialize(string base64OperationId)
+        {
+            byte[] plainTextBytes = Convert.FromBase64String(base64OperationId);
+
+            var options = new JsonSerializerOptions() { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
```

Comment:
Should we consider declaring this set of options as a `static readonly` since they're stateless and used in several places?

Reply:
Good idea. We can make sure they're kept in sync this way.